### PR TITLE
Created script for converting between US and UK symbol names

### PR DESCRIPTION
**Pull request by DavidSM64**
The style guide now requires that UK English names be used for symbols. (eg: Use `colour` instead of `color`)

To help with this I created a python script to automatically search and replace all symbol names, except for those in the `/lib/` directory. The script can also be used to convert all the names back into US English if we ever change our minds in the future.
Usage: 
* US to UK names -> `python3 tools/python/fix_regional_names.py`
* UK to US names -> `python3 tools/python/fix_regional_names.py -c us`

By default, version `us_1.0` is assumed. You can change this by adding the `-v` argument like this: `-v eu_1.1`. Although this won't work yet, as we currently only support `us_1.0` at the moment.

### CODE REVIEW

File: tools/python/fix_regional_names.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+diffMap = json.loads(FileUtil.get_text_from_file(REGIONAL_NAMES_JSON_PATH))['diffMap']
+
+def precheckForErrors(version):
+    versionBuildPath = 'build/' + version
+    if not FileUtil.does_file_exist(versionBuildPath):
+        print('Error: "' + versionBuildPath + '" could not be found. You should have an OK build before running this script!')
+        return None
+    symMapPath = versionBuildPath + '/dkr.map'
+    if not FileUtil.does_file_exist(symMapPath):
+        print('Error: "' + symMapPath + '" could not be found. Make sure this script is running from the root directory of the repo.')
```
[the remark this answers is not in the document]
Comment (DavidSM64) [reply tools/python/fix_regional_names.py]: Fixed

File: tools/python/fix_regional_names.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    for i in range(0, len(lines)):
+        line = lines[i]
+        if state == 0:
+            if line.startswith('.main'):
+                state = 1
+        elif state == 1:
+            if line.startswith(' build/'):
+                if '/lib/' not in line:
+                    state = 2
```
[the remark this answers is not in the document]
Comment (DavidSM64) [reply tools/python/fix_regional_names.py]: Fixed

File: tools/python/fix_regional_names.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        elif state == 2:
+            if line.startswith(' build/'):
+                if '/lib/' in line:
+                    state = 1
+                    continue
```
[the remark this answers is not in the document]
Comment (DavidSM64) [reply tools/python/fix_regional_names.py]: Fixed

File: tools/python/fix_regional_names.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    validSymbols = {}
+
+    for symbol in symbols:
+        symbolParts = breakDownSymbol(symbol)
+
+        if symbolParts is None:
+            # Symbol can't be processed, so just skip it.
+            continue
+
+        for i in range(0, len(diffMap)):
```
[the remark this answers is not in the document]
Comment (DavidSM64) [reply tools/python/fix_regional_names.py]: Good call. I Implemented this change and now there is no delay. 👍

File: tools/python/fix_regional_names.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+            elif response.startswith('n'):
+                print("\033[1A\033[" + str(len(inputText)) + "C" + TERMINAL_COLOR_RED + "Change Discarded" + TERMINAL_COLOR_RESET)
+                break
+            elif response.startswith('x'):
+                return None # None means aborted
+        
+    return validatedReplaces 
+
+def convertName(convertTo, version):
+    #print("Parameters: " + convertTo + ", " + version)
```
[the remark this answers is not in the document]
Comment (DavidSM64) [reply tools/python/fix_regional_names.py]: Fixed

File: lib/asm/exception/__osException.s [hunk 6]
```diff
@@ -191,8 +191,8 @@ glabel __osException
 /* 0D3B2C 800D2F2C 3C01800E */  lui   $at, %hi(__osHwIntTable) # $at, 0x800e
 /* 0D3B30 800D2F30 002A0821 */  addu  $at, $at, $t2
 /* 0D3B34 800D2F34 8C2A48A0 */  lw    $t2, %lo(__osHwIntTable)($at)
-/* 0D3B38 800D2F38 3C1D8013 */  lui   $sp, %hi(leoDiskStack) # $sp, 0x8013
-/* 0D3B3C 800D2F3C 27BDAAE0 */  addiu $sp, %lo(leoDiskStack) # addiu $sp, $sp, -0x5520
+/* 0D3B38 800D2F38 3C1D8013 */  lui   $sp, %hi(leoDiscStack) # $sp, 0x8013
```
Comment (Ryan-Myers) [inline lib/asm/exception/__osException.s:194]: This might be an exception. In computing terms  I believe we want disk here instead.

**Review by Ryan-Myers — APPROVED** [review]
All looks good to me except leoDiskStack.

File: lib/asm/exception/__osException.s (hunk 6 above)
Comment (DavidSM64) [reply lib/asm/exception/__osException.s]: Fixed

**Pull request merged**